src/into_dart.rs

```diff
@@ -185,7 +185,7 @@ macro_rules! dart_typed_data_type_trait_impl {
             #[cfg(feature="zero-copy")]
             impl IntoDart for Vec<$rust_type> {
                 fn into_dart(self) -> DartCObject {
-                    let mut vec = ManuallyDrop::new(self.0);
+                    let mut vec = ManuallyDrop::new(self);
```

Comment:
I did not check the code in the last PR. By checking it, it seems we are having some copy-paste here (e.g. this dozen of lines vs ZeroCopyBuffer code). Is it possible we extract it out? For example, extract to a separate function? Then, we will not be faced with such errors.